// node-lyft/lib/model/ApiError.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _ErrorDetail = require('./ErrorDetail');

var _ErrorDetail2 = _interopRequireDefault(_ErrorDetail);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The ApiError model module.
* @module model/ApiError
* @version 1.0.0
*/
var ApiError = function () {
    /**
    * Constructs a new <code>ApiError</code>.
    * Details about why a request failed, such as missing or invalid parameters
    * @alias module:model/ApiError
    * @class
    */

    function ApiError() {
        _classCallCheck(this, ApiError);

        this.error = undefined;
        this.error_detail = undefined;
        this.error_description = undefined;
    }

    /**
    * Constructs a <code>ApiError</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/ApiError} obj Optional instance to populate.
    * @return {module:model/ApiError} The populated <code>ApiError</code> instance.
    */


    _createClass(ApiError, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new ApiError();

                if (data.hasOwnProperty('error')) {
                    obj['error'] = _ApiClient2.default.convertToType(data['error'], 'String');
                }
                if (data.hasOwnProperty('error_detail')) {
                    obj['error_detail'] = _ApiClient2.default.convertToType(data['error_detail'], [_ErrorDetail2.default]);
                }
                if (data.hasOwnProperty('error_description')) {
                    obj['error_description'] = _ApiClient2.default.convertToType(data['error_description'], 'String');
                }
            }
            return obj;
        }

        /**
        * A \"slug\" that serves as the error code (eg. \"bad_parameter\")
        * @member {String} error
        */

        /**
        * @member {Array.<module:model/ErrorDetail>} error_detail
        */

        /**
        * A user-friendly description of the error (appropriate to show to an end-user)
        * @member {String} error_description
        */

    }]);

    return ApiError;
}();

exports.default = ApiError;
