// node-lyft/lib/model/CancellationCostError.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _ApiError = require('./ApiError');

var _ApiError2 = _interopRequireDefault(_ApiError);

var _CancellationCost = require('./CancellationCost');

var _CancellationCost2 = _interopRequireDefault(_CancellationCost);

var _ErrorDetail = require('./ErrorDetail');

var _ErrorDetail2 = _interopRequireDefault(_ErrorDetail);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The CancellationCostError model module.
* @module model/CancellationCostError
* @version 1.0.0
*/
var CancellationCostError = function () {
    /**
    * Constructs a new <code>CancellationCostError</code>.
    * @alias module:model/CancellationCostError
    * @class
    * @implements module:model/CancellationCost
    * @implements module:model/ApiError
    * @param amount {Number} Total price of the ride
    * @param currency {String} The ISO 4217 currency code for the amount (e.g. USD)
    * @param description {String} The description for the cost
    */

    function CancellationCostError(amount, currency, description) {
        _classCallCheck(this, CancellationCostError);

        this.amount = undefined;
        this.currency = undefined;
        this.description = undefined;
        this.token = undefined;
        this.token_duration = undefined;
        this.error = undefined;
        this.error_detail = undefined;
        this.error_description = undefined;


        _CancellationCost2.default.call(this, amount, currency, description);_ApiError2.default.call(this);
    }

    /**
    * Constructs a <code>CancellationCostError</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/CancellationCostError} obj Optional instance to populate.
    * @return {module:model/CancellationCostError} The populated <code>CancellationCostError</code> instance.
    */


    _createClass(CancellationCostError, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new CancellationCostError();

                _CancellationCost2.default.constructFromObject(data, obj);_ApiError2.default.constructFromObject(data, obj);
            }
            return obj;
        }

        // Implement CancellationCost interface:
        /**
        * Total price of the ride
        * @member {Number} amount
        */

        /**
            * The ISO 4217 currency code for the amount (e.g. USD)
            * @member {String} currency
            */

        /**
            * The description for the cost
            * @member {String} description
            */

        /**
            * Token used to confirm the fee when cancelling a request
            * @member {String} token
            */

        /**
            * How long, in seconds, before the token expires
            * @member {Number} token_duration
            */


        // Implement ApiError interface:
        /**
        * A \"slug\" that serves as the error code (eg. \"bad_parameter\")
        * @member {String} error
        */

        /**
            * @member {Array.<module:model/ErrorDetail>} error_detail
            */

        /**
            * A user-friendly description of the error (appropriate to show to an end-user)
            * @member {String} error_description
            */

    }]);

    return CancellationCostError;
}();

exports.default = CancellationCostError;
