// node-lyft/lib/model/CancellationCost.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _Cost = require('./Cost');

var _Cost2 = _interopRequireDefault(_Cost);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The CancellationCost model module.
* @module model/CancellationCost
* @version 1.0.0
*/
var CancellationCost = function () {
    /**
    * Constructs a new <code>CancellationCost</code>.
    * @alias module:model/CancellationCost
    * @class
    * @implements module:model/Cost
    * @param amount {Number} Total price of the ride
    * @param currency {String} The ISO 4217 currency code for the amount (e.g. USD)
    * @param description {String} The description for the cost
    */

    function CancellationCost(amount, currency, description) {
        _classCallCheck(this, CancellationCost);

        this.token = undefined;
        this.token_duration = undefined;
        this.amount = undefined;
        this.currency = undefined;
        this.description = undefined;


        _Cost2.default.call(this, amount, currency, description);
    }

    /**
    * Constructs a <code>CancellationCost</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/CancellationCost} obj Optional instance to populate.
    * @return {module:model/CancellationCost} The populated <code>CancellationCost</code> instance.
    */


    _createClass(CancellationCost, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new CancellationCost();

                _Cost2.default.constructFromObject(data, obj);

                if (data.hasOwnProperty('token')) {
                    obj['token'] = _ApiClient2.default.convertToType(data['token'], 'String');
                }
                if (data.hasOwnProperty('token_duration')) {
                    obj['token_duration'] = _ApiClient2.default.convertToType(data['token_duration'], 'Number');
                }
            }
            return obj;
        }

        /**
        * Token used to confirm the fee when cancelling a request
        * @member {String} token
        */

        /**
        * How long, in seconds, before the token expires
        * @member {Number} token_duration
        */


        // Implement Cost interface:
        /**
        * Total price of the ride
        * @member {Number} amount
        */

        /**
            * The ISO 4217 currency code for the amount (e.g. USD)
            * @member {String} currency
            */

        /**
            * The description for the cost
            * @member {String} description
            */

    }]);

    return CancellationCost;
}();

exports.default = CancellationCost;
