// node-lyft/lib/model/CostEstimate.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _RideTypeEnum = require('./RideTypeEnum');

var _RideTypeEnum2 = _interopRequireDefault(_RideTypeEnum);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The CostEstimate model module.
* @module model/CostEstimate
* @version 1.0.0
*/
var CostEstimate = function () {
    /**
    * Constructs a new <code>CostEstimate</code>.
    * A non-guaranteed estimate of price
    * @alias module:model/CostEstimate
    * @class
    */

    function CostEstimate() {
        _classCallCheck(this, CostEstimate);

        this.ride_type = undefined;
        this.display_name = undefined;
        this.currency = undefined;
        this.estimated_cost_cents_min = undefined;
        this.estimated_cost_cents_max = undefined;
        this.estimated_distance_miles = undefined;
        this.estimated_duration_seconds = undefined;
        this.is_valid_estimate = undefined;
        this.primetime_percentage = undefined;
        this.primetime_confirmation_token = undefined;
        this.cost_token = undefined;
    }

    /**
    * Constructs a <code>CostEstimate</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/CostEstimate} obj Optional instance to populate.
    * @return {module:model/CostEstimate} The populated <code>CostEstimate</code> instance.
    */


    _createClass(CostEstimate, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new CostEstimate();

                if (data.hasOwnProperty('ride_type')) {
                    obj['ride_type'] = _RideTypeEnum2.default.constructFromObject(data['ride_type']);
                }
                if (data.hasOwnProperty('display_name')) {
                    obj['display_name'] = _ApiClient2.default.convertToType(data['display_name'], 'String');
                }
                if (data.hasOwnProperty('currency')) {
                    obj['currency'] = _ApiClient2.default.convertToType(data['currency'], 'String');
                }
                if (data.hasOwnProperty('estimated_cost_cents_min')) {
                    obj['estimated_cost_cents_min'] = _ApiClient2.default.convertToType(data['estimated_cost_cents_min'], 'Number');
                }
                if (data.hasOwnProperty('estimated_cost_cents_max')) {
                    obj['estimated_cost_cents_max'] = _ApiClient2.default.convertToType(data['estimated_cost_cents_max'], 'Number');
                }
                if (data.hasOwnProperty('estimated_distance_miles')) {
                    obj['estimated_distance_miles'] = _ApiClient2.default.convertToType(data['estimated_distance_miles'], 'Number');
                }
                if (data.hasOwnProperty('estimated_duration_seconds')) {
                    obj['estimated_duration_seconds'] = _ApiClient2.default.convertToType(data['estimated_duration_seconds'], 'Number');
                }
                if (data.hasOwnProperty('is_valid_estimate')) {
                    obj['is_valid_estimate'] = _ApiClient2.default.convertToType(data['is_valid_estimate'], 'Boolean');
                }
                if (data.hasOwnProperty('primetime_percentage')) {
                    obj['primetime_percentage'] = _ApiClient2.default.convertToType(data['primetime_percentage'], 'String');
                }
                if (data.hasOwnProperty('primetime_confirmation_token')) {
                    obj['primetime_confirmation_token'] = _ApiClient2.default.convertToType(data['primetime_confirmation_token'], 'String');
                }
                if (data.hasOwnProperty('cost_token')) {
                    obj['cost_token'] = _ApiClient2.default.convertToType(data['cost_token'], 'String');
                }
            }
            return obj;
        }

        /**
        * @member {module:model/RideTypeEnum} ride_type
        */

        /**
        * A human readable description of the ride type
        * @member {String} display_name
        */

        /**
        * The ISO 4217 currency code for the amount (e.g. 'USD')
        * @member {String} currency
        */

        /**
        * Estimated lower bound for trip cost, in minor units (cents). Estimates are not guaranteed, and only provide a reasonable range based on current conditions. 
        * @member {Number} estimated_cost_cents_min
        */

        /**
        * Estimated upper bound for trip cost, in minor units (cents). Estimates are not guaranteed, and only provide a reasonable range based on current conditions. 
        * @member {Number} estimated_cost_cents_max
        */

        /**
        * Estimated distance for this trip 
        * @member {Number} estimated_distance_miles
        */

        /**
        * Estimated time to get from the start location to the end. 
        * @member {Number} estimated_duration_seconds
        */

        /**
        * The validity of the cost estimate returned
        * @member {Boolean} is_valid_estimate
        */

        /**
        * Current Prime Time Percentage. Prime Time adds a percentage to ride costs, prior to other applicable fees. When ride requests greatly outnumber available drivers, our system will automatically turn on Prime Time. If Prime Time is inactive, the value returned will be '0%'. Note: The returned estimate already has Prime Time factored in. The value is returned here for reference and to allow users to confirm/accept Prime Time prior to initiating a ride. 
        * @member {String} primetime_percentage
        */

        /**
        * This token is needed when requesting rides. (Deprecated)
        * @member {String} primetime_confirmation_token
        */

        /**
        * A token that confirms the user has accepted current Prime Time and/or fixed price charges. See 'Request a Lyft' for more details
        * @member {String} cost_token
        */

    }]);

    return CostEstimate;
}();

exports.default = CostEstimate;
