// node-lyft/lib/model/CurrentRideLocation.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _LatLng = require('./LatLng');

var _LatLng2 = _interopRequireDefault(_LatLng);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The CurrentRideLocation model module.
* @module model/CurrentRideLocation
* @version 1.0.0
*/
var CurrentRideLocation = function () {
    /**
    * Constructs a new <code>CurrentRideLocation</code>.
    * @alias module:model/CurrentRideLocation
    * @class
    * @implements module:model/LatLng
    * @param lat {Number} The latitude component of a location
    * @param lng {Number} The longitude component of a location
    */

    function CurrentRideLocation(lat, lng) {
        _classCallCheck(this, CurrentRideLocation);

        this.bearing = undefined;
        this.lat = undefined;
        this.lng = undefined;


        _LatLng2.default.call(this, lat, lng);
    }

    /**
    * Constructs a <code>CurrentRideLocation</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/CurrentRideLocation} obj Optional instance to populate.
    * @return {module:model/CurrentRideLocation} The populated <code>CurrentRideLocation</code> instance.
    */


    _createClass(CurrentRideLocation, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new CurrentRideLocation();

                _LatLng2.default.constructFromObject(data, obj);

                if (data.hasOwnProperty('bearing')) {
                    obj['bearing'] = _ApiClient2.default.convertToType(data['bearing'], 'Number');
                }
            }
            return obj;
        }

        /**
        * Bearing of the driver's car in degrees
        * @member {Number} bearing
        */


        // Implement LatLng interface:
        /**
        * The latitude component of a location
        * @member {Number} lat
        */

        /**
            * The longitude component of a location
            * @member {Number} lng
            */

    }]);

    return CurrentRideLocation;
}();

exports.default = CurrentRideLocation;
