// node-lyft/lib/model/DriverDetail.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The DriverDetail model module.
* @module model/DriverDetail
* @version 1.0.0
*/
var DriverDetail = function () {
    /**
    * Constructs a new <code>DriverDetail</code>.
    * @alias module:model/DriverDetail
    * @class
    * @param firstName {String} The driver's first name
    * @param phoneNumber {String} The driver's contact phone number. Must be E.164 formatted. 
    * @param rating {String} The driver's rating based in 0-5 scale
    * @param imageUrl {String} The driver's image url
    * @param userId {String} The driver's id
    */

    function DriverDetail(firstName, phoneNumber, rating, imageUrl, userId) {
        _classCallCheck(this, DriverDetail);

        this.first_name = undefined;
        this.phone_number = undefined;
        this.rating = undefined;
        this.image_url = undefined;
        this.user_id = undefined;


        this['first_name'] = firstName;this['phone_number'] = phoneNumber;this['rating'] = rating;this['image_url'] = imageUrl;this['user_id'] = userId;
    }

    /**
    * Constructs a <code>DriverDetail</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/DriverDetail} obj Optional instance to populate.
    * @return {module:model/DriverDetail} The populated <code>DriverDetail</code> instance.
    */


    _createClass(DriverDetail, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new DriverDetail();

                if (data.hasOwnProperty('first_name')) {
                    obj['first_name'] = _ApiClient2.default.convertToType(data['first_name'], 'String');
                }
                if (data.hasOwnProperty('phone_number')) {
                    obj['phone_number'] = _ApiClient2.default.convertToType(data['phone_number'], 'String');
                }
                if (data.hasOwnProperty('rating')) {
                    obj['rating'] = _ApiClient2.default.convertToType(data['rating'], 'String');
                }
                if (data.hasOwnProperty('image_url')) {
                    obj['image_url'] = _ApiClient2.default.convertToType(data['image_url'], 'String');
                }
                if (data.hasOwnProperty('user_id')) {
                    obj['user_id'] = _ApiClient2.default.convertToType(data['user_id'], 'String');
                }
            }
            return obj;
        }

        /**
        * The driver's first name
        * @member {String} first_name
        */

        /**
        * The driver's contact phone number. Must be E.164 formatted. 
        * @member {String} phone_number
        */

        /**
        * The driver's rating based in 0-5 scale
        * @member {String} rating
        */

        /**
        * The driver's image url
        * @member {String} image_url
        */

        /**
        * The driver's id
        * @member {String} user_id
        */

    }]);

    return DriverDetail;
}();

exports.default = DriverDetail;
