// node-lyft/lib/model/Eta.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _RideTypeEnum = require('./RideTypeEnum');

var _RideTypeEnum2 = _interopRequireDefault(_RideTypeEnum);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The Eta model module.
* @module model/Eta
* @version 1.0.0
*/
var Eta = function () {
    /**
    * Constructs a new <code>Eta</code>.
    * Estimated Time of Arrival
    * @alias module:model/Eta
    * @class
    */

    function Eta() {
        _classCallCheck(this, Eta);

        this.ride_type = undefined;
        this.display_name = undefined;
        this.eta_seconds = undefined;
        this.eta_seconds_max = undefined;
        this.is_valid_estimate = undefined;
    }

    /**
    * Constructs a <code>Eta</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Eta} obj Optional instance to populate.
    * @return {module:model/Eta} The populated <code>Eta</code> instance.
    */


    _createClass(Eta, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new Eta();

                if (data.hasOwnProperty('ride_type')) {
                    obj['ride_type'] = _RideTypeEnum2.default.constructFromObject(data['ride_type']);
                }
                if (data.hasOwnProperty('display_name')) {
                    obj['display_name'] = _ApiClient2.default.convertToType(data['display_name'], 'String');
                }
                if (data.hasOwnProperty('eta_seconds')) {
                    obj['eta_seconds'] = _ApiClient2.default.convertToType(data['eta_seconds'], 'Number');
                }
                if (data.hasOwnProperty('eta_seconds_max')) {
                    obj['eta_seconds_max'] = _ApiClient2.default.convertToType(data['eta_seconds_max'], 'Number');
                }
                if (data.hasOwnProperty('is_valid_estimate')) {
                    obj['is_valid_estimate'] = _ApiClient2.default.convertToType(data['is_valid_estimate'], 'Boolean');
                }
            }
            return obj;
        }

        /**
        * @member {module:model/RideTypeEnum} ride_type
        */

        /**
        * A human readable description of the ride type
        * @member {String} display_name
        */

        /**
        * Estimated seconds for a driver to arrive
        * @member {Number} eta_seconds
        */

        /**
        * Estimated upper bound of seconds for a driver to arrive
        * @member {Number} eta_seconds_max
        */

        /**
        * The validity of the ETA estimate returned
        * @member {Boolean} is_valid_estimate
        */

    }]);

    return Eta;
}();

exports.default = Eta;
