// node-lyft/lib/model/LineItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The LineItem model module.
* @module model/LineItem
* @version 1.0.0
*/
var LineItem = function () {
    /**
    * Constructs a new <code>LineItem</code>.
    * @alias module:model/LineItem
    * @class
    * @param type {String} The line item display name for a charge item
    * @param amount {Number} The line item amount
    * @param currency {String} The currency for the amount
    */

    function LineItem(type, amount, currency) {
        _classCallCheck(this, LineItem);

        this.type = undefined;
        this.amount = undefined;
        this.currency = undefined;


        this['type'] = type;this['amount'] = amount;this['currency'] = currency;
    }

    /**
    * Constructs a <code>LineItem</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/LineItem} obj Optional instance to populate.
    * @return {module:model/LineItem} The populated <code>LineItem</code> instance.
    */


    _createClass(LineItem, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new LineItem();

                if (data.hasOwnProperty('type')) {
                    obj['type'] = _ApiClient2.default.convertToType(data['type'], 'String');
                }
                if (data.hasOwnProperty('amount')) {
                    obj['amount'] = _ApiClient2.default.convertToType(data['amount'], 'Number');
                }
                if (data.hasOwnProperty('currency')) {
                    obj['currency'] = _ApiClient2.default.convertToType(data['currency'], 'String');
                }
            }
            return obj;
        }

        /**
        * The line item display name for a charge item
        * @member {String} type
        */

        /**
        * The line item amount
        * @member {Number} amount
        */

        /**
        * The currency for the amount
        * @member {String} currency
        */

    }]);

    return LineItem;
}();

exports.default = LineItem;
