// node-lyft/lib/model/PassengerDetail.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _UserDetail = require('./UserDetail');

var _UserDetail2 = _interopRequireDefault(_UserDetail);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The PassengerDetail model module.
* @module model/PassengerDetail
* @version 1.0.0
*/
var PassengerDetail = function () {
    /**
    * Constructs a new <code>PassengerDetail</code>.
    * @alias module:model/PassengerDetail
    * @class
    * @implements module:model/UserDetail
    * @param firstName {String} The passenger's first name
    * @param imageUrl {String} The passenger's profile image
    * @param rating {String} The passenger's rating
    */

    function PassengerDetail(firstName, imageUrl, rating) {
        _classCallCheck(this, PassengerDetail);

        this.last_name = undefined;
        this.user_id = undefined;
        this.first_name = undefined;
        this.image_url = undefined;
        this.rating = undefined;


        _UserDetail2.default.call(this, firstName, imageUrl, rating);
    }

    /**
    * Constructs a <code>PassengerDetail</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/PassengerDetail} obj Optional instance to populate.
    * @return {module:model/PassengerDetail} The populated <code>PassengerDetail</code> instance.
    */


    _createClass(PassengerDetail, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new PassengerDetail();

                _UserDetail2.default.constructFromObject(data, obj);

                if (data.hasOwnProperty('last_name')) {
                    obj['last_name'] = _ApiClient2.default.convertToType(data['last_name'], 'String');
                }
                if (data.hasOwnProperty('user_id')) {
                    obj['user_id'] = _ApiClient2.default.convertToType(data['user_id'], 'String');
                }
            }
            return obj;
        }

        /**
        * The passenger's last name
        * @member {String} last_name
        */

        /**
        * The passenger's lyft user id
        * @member {String} user_id
        */


        // Implement UserDetail interface:
        /**
        * The passenger's first name
        * @member {String} first_name
        */

        /**
            * The passenger's profile image
            * @member {String} image_url
            */

        /**
            * The passenger's rating
            * @member {String} rating
            */

    }]);

    return PassengerDetail;
}();

exports.default = PassengerDetail;
