// node-lyft/lib/model/PickupDropoffLocation.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _Location = require('./Location');

var _Location2 = _interopRequireDefault(_Location);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The PickupDropoffLocation model module.
* @module model/PickupDropoffLocation
* @version 1.0.0
*/
var PickupDropoffLocation = function () {
    /**
    * Constructs a new <code>PickupDropoffLocation</code>.
    * @alias module:model/PickupDropoffLocation
    * @class
    * @implements module:model/Location
    * @param lat {Number} The latitude component of a location
    * @param lng {Number} The longitude component of a location
    */

    function PickupDropoffLocation(lat, lng) {
        _classCallCheck(this, PickupDropoffLocation);

        this.time = undefined;
        this.lat = undefined;
        this.lng = undefined;
        this.address = undefined;


        _Location2.default.call(this, lat, lng);
    }

    /**
    * Constructs a <code>PickupDropoffLocation</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/PickupDropoffLocation} obj Optional instance to populate.
    * @return {module:model/PickupDropoffLocation} The populated <code>PickupDropoffLocation</code> instance.
    */


    _createClass(PickupDropoffLocation, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new PickupDropoffLocation();

                _Location2.default.constructFromObject(data, obj);

                if (data.hasOwnProperty('time')) {
                    obj['time'] = _ApiClient2.default.convertToType(data['time'], 'Date');
                }
            }
            return obj;
        }

        /**
        * Server time when the location object is created
        * @member {Date} time
        */


        // Implement Location interface:
        /**
        * The latitude component of a location
        * @member {Number} lat
        */

        /**
            * The longitude component of a location
            * @member {Number} lng
            */

        /**
            * A human readable address at/near the given location
            * @member {String} address
            */

    }]);

    return PickupDropoffLocation;
}();

exports.default = PickupDropoffLocation;
