// node-lyft/lib/model/PricingDetails.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The PricingDetails model module.
* @module model/PricingDetails
* @version 1.0.0
*/
var PricingDetails = function () {
    /**
    * Constructs a new <code>PricingDetails</code>.
    * @alias module:model/PricingDetails
    * @class
    */

    function PricingDetails() {
        _classCallCheck(this, PricingDetails);

        this.base_charge = undefined;
        this.cancel_penalty_amount = undefined;
        this.cost_minimum = undefined;
        this.cost_per_mile = undefined;
        this.cost_per_minute = undefined;
        this.currency = undefined;
        this.trust_and_service = undefined;
    }

    /**
    * Constructs a <code>PricingDetails</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/PricingDetails} obj Optional instance to populate.
    * @return {module:model/PricingDetails} The populated <code>PricingDetails</code> instance.
    */


    _createClass(PricingDetails, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new PricingDetails();

                if (data.hasOwnProperty('base_charge')) {
                    obj['base_charge'] = _ApiClient2.default.convertToType(data['base_charge'], 'Number');
                }
                if (data.hasOwnProperty('cancel_penalty_amount')) {
                    obj['cancel_penalty_amount'] = _ApiClient2.default.convertToType(data['cancel_penalty_amount'], 'Number');
                }
                if (data.hasOwnProperty('cost_minimum')) {
                    obj['cost_minimum'] = _ApiClient2.default.convertToType(data['cost_minimum'], 'Number');
                }
                if (data.hasOwnProperty('cost_per_mile')) {
                    obj['cost_per_mile'] = _ApiClient2.default.convertToType(data['cost_per_mile'], 'Number');
                }
                if (data.hasOwnProperty('cost_per_minute')) {
                    obj['cost_per_minute'] = _ApiClient2.default.convertToType(data['cost_per_minute'], 'Number');
                }
                if (data.hasOwnProperty('currency')) {
                    obj['currency'] = _ApiClient2.default.convertToType(data['currency'], 'String');
                }
                if (data.hasOwnProperty('trust_and_service')) {
                    obj['trust_and_service'] = _ApiClient2.default.convertToType(data['trust_and_service'], 'Number');
                }
            }
            return obj;
        }

        /**
        * The base charge of the trip
        * @member {Number} base_charge
        */

        /**
        * The charge amount if cancel penalty is involved
        * @member {Number} cancel_penalty_amount
        */

        /**
        * The minimum charge for the trip
        * @member {Number} cost_minimum
        */

        /**
        * The cost per mile
        * @member {Number} cost_per_mile
        */

        /**
        * The cost per minute
        * @member {Number} cost_per_minute
        */

        /**
        * The ISO 4217 currency code for the amount (e.g. USD)
        * @member {String} currency
        */

        /**
        * Service fee
        * @member {Number} trust_and_service
        */

    }]);

    return PricingDetails;
}();

exports.default = PricingDetails;
