// node-lyft/lib/model/RatingRequest.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The RatingRequest model module.
* @module model/RatingRequest
* @version 1.0.0
*/
var RatingRequest = function () {
    /**
    * Constructs a new <code>RatingRequest</code>.
    * Rating and optional feedback and tip
    * @alias module:model/RatingRequest
    * @class
    * @param rating {Number} The passenger's rating of this ride from 1 to 5
    */

    function RatingRequest(rating) {
        _classCallCheck(this, RatingRequest);

        this.rating = undefined;
        this.feedback = undefined;
        this.tip = undefined;


        this['rating'] = rating;
    }

    /**
    * Constructs a <code>RatingRequest</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/RatingRequest} obj Optional instance to populate.
    * @return {module:model/RatingRequest} The populated <code>RatingRequest</code> instance.
    */


    _createClass(RatingRequest, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new RatingRequest();

                if (data.hasOwnProperty('rating')) {
                    obj['rating'] = _ApiClient2.default.convertToType(data['rating'], 'Number');
                }
                if (data.hasOwnProperty('feedback')) {
                    obj['feedback'] = _ApiClient2.default.convertToType(data['feedback'], 'String');
                }
                if (data.hasOwnProperty('tip')) {
                    obj['tip'] = _ApiClient2.default.convertToType(data['tip'], Object);
                }
            }
            return obj;
        }

        /**
        * The passenger's rating of this ride from 1 to 5
        * @member {Number} rating
        */

        /**
        * The passenger's written feedback about this ride
        * @member {String} feedback
        */

        /**
        * Tip amount in minor units and tip currency
        * @member {Object} tip
        */

    }]);

    return RatingRequest;
}();

exports.default = RatingRequest;
