// node-lyft/lib/model/Ride.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _RideTypeEnum = require('./RideTypeEnum');

var _RideTypeEnum2 = _interopRequireDefault(_RideTypeEnum);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The Ride model module.
* @module model/Ride
* @version 1.0.0
*/
var Ride = function () {
    /**
    * Constructs a new <code>Ride</code>.
    * Represents a requested, ongoing, or finished Lyft ride
    * @alias module:model/Ride
    * @class
    * @param rideType {module:model/RideTypeEnum} 
    * @param origin {Object} The *requested* location for passenger pickup
    */

    function Ride(rideType, origin) {
        _classCallCheck(this, Ride);

        this.ride_type = undefined;
        this.origin = undefined;
        this.destination = undefined;
        this.primetime_confirmation_token = undefined;
        this.cost_token = undefined;


        this['ride_type'] = rideType;this['origin'] = origin;
    }

    /**
    * Constructs a <code>Ride</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/Ride} obj Optional instance to populate.
    * @return {module:model/Ride} The populated <code>Ride</code> instance.
    */


    _createClass(Ride, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new Ride();

                if (data.hasOwnProperty('ride_type')) {
                    obj['ride_type'] = _RideTypeEnum2.default.constructFromObject(data['ride_type']);
                }
                if (data.hasOwnProperty('origin')) {
                    obj['origin'] = _ApiClient2.default.convertToType(data['origin'], Object);
                }
                if (data.hasOwnProperty('destination')) {
                    obj['destination'] = _ApiClient2.default.convertToType(data['destination'], Object);
                }
                if (data.hasOwnProperty('primetime_confirmation_token')) {
                    obj['primetime_confirmation_token'] = _ApiClient2.default.convertToType(data['primetime_confirmation_token'], 'String');
                }
                if (data.hasOwnProperty('cost_token')) {
                    obj['cost_token'] = _ApiClient2.default.convertToType(data['cost_token'], 'String');
                }
            }
            return obj;
        }

        /**
        * @member {module:model/RideTypeEnum} ride_type
        */

        /**
        * The *requested* location for passenger pickup
        * @member {Object} origin
        */

        /**
        * The *requested* location for passenger drop off
        * @member {Object} destination
        */

        /**
        * A token that confirms the user has accepted current primetime charges (Deprecated)
        * @member {String} primetime_confirmation_token
        */

        /**
        * A token that confirms the user has accepted current Prime Time and/or fixed price charges
        * @member {String} cost_token
        */

    }]);

    return Ride;
}();

exports.default = Ride;
