// node-lyft/lib/model/RideStatusEnum.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require("../ApiClient");

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* Enum class RideStatusEnum.
* @enum {}
* @readonly
*/
var RideStatusEnum = function () {
    function RideStatusEnum() {
        _classCallCheck(this, RideStatusEnum);

        this.pending = "pending";
        this.accepted = "accepted";
        this.arrived = "arrived";
        this.pickedUp = "pickedUp";
        this.droppedOff = "droppedOff";
        this.canceled = "canceled";
        this.scheduled = "scheduled";
        this.unknown = "unknown";
    }

    /**
     * value: "pending"
     * @const
     */


    /**
     * value: "accepted"
     * @const
     */


    /**
     * value: "arrived"
     * @const
     */


    /**
     * value: "pickedUp"
     * @const
     */


    /**
     * value: "droppedOff"
     * @const
     */


    /**
     * value: "canceled"
     * @const
     */


    /**
     * value: "scheduled"
     * @const
     */


    /**
     * value: "unknown"
     * @const
     */


    _createClass(RideStatusEnum, null, [{
        key: "constructFromObject",


        /**
        * Returns a <code>RideStatusEnum</code> enum value from a Javascript object name.
        * @param {Object} data The plain JavaScript object containing the name of the enum value.
        * @return {module:model/RideStatusEnum} The enum <code>RideStatusEnum</code> value.
        */
        value: function constructFromObject(object) {
            return object;
        }
    }]);

    return RideStatusEnum;
}();

exports.default = RideStatusEnum;
