// node-lyft/lib/model/RideRequestError.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _ErrorDetail = require('./ErrorDetail');

var _ErrorDetail2 = _interopRequireDefault(_ErrorDetail);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The RideRequestError model module.
* @module model/RideRequestError
* @version 1.0.0
*/
var RideRequestError = function () {
    /**
    * Constructs a new <code>RideRequestError</code>.
    * Details about why a request failed, such as missing or invalid parameters
    * @alias module:model/RideRequestError
    * @class
    * @param error {String} A \"slug\" that serves as the error code (eg. \"bad_parameter\")
    */

    function RideRequestError(error) {
        _classCallCheck(this, RideRequestError);

        this.error = undefined;
        this.error_detail = undefined;
        this.error_description = undefined;
        this.primetime_percentage = undefined;
        this.primetime_multiplier = undefined;
        this.primetime_confirmation_token = undefined;
        this.cost_token = undefined;
        this.token_duration = undefined;


        this['error'] = error;
    }

    /**
    * Constructs a <code>RideRequestError</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/RideRequestError} obj Optional instance to populate.
    * @return {module:model/RideRequestError} The populated <code>RideRequestError</code> instance.
    */


    _createClass(RideRequestError, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new RideRequestError();

                if (data.hasOwnProperty('error')) {
                    obj['error'] = _ApiClient2.default.convertToType(data['error'], 'String');
                }
                if (data.hasOwnProperty('error_detail')) {
                    obj['error_detail'] = _ApiClient2.default.convertToType(data['error_detail'], [_ErrorDetail2.default]);
                }
                if (data.hasOwnProperty('error_description')) {
                    obj['error_description'] = _ApiClient2.default.convertToType(data['error_description'], 'String');
                }
                if (data.hasOwnProperty('primetime_percentage')) {
                    obj['primetime_percentage'] = _ApiClient2.default.convertToType(data['primetime_percentage'], 'String');
                }
                if (data.hasOwnProperty('primetime_multiplier')) {
                    obj['primetime_multiplier'] = _ApiClient2.default.convertToType(data['primetime_multiplier'], 'Number');
                }
                if (data.hasOwnProperty('primetime_confirmation_token')) {
                    obj['primetime_confirmation_token'] = _ApiClient2.default.convertToType(data['primetime_confirmation_token'], 'String');
                }
                if (data.hasOwnProperty('cost_token')) {
                    obj['cost_token'] = _ApiClient2.default.convertToType(data['cost_token'], 'String');
                }
                if (data.hasOwnProperty('token_duration')) {
                    obj['token_duration'] = _ApiClient2.default.convertToType(data['token_duration'], 'String');
                }
            }
            return obj;
        }

        /**
        * A \"slug\" that serves as the error code (eg. \"bad_parameter\")
        * @member {String} error
        */

        /**
        * @member {Array.<module:model/ErrorDetail>} error_detail
        */

        /**
        * A user-friendly description of the error (appropriate to show to an end-user)
        * @member {String} error_description
        */

        /**
        * Current Prime Time percentage
        * @member {String} primetime_percentage
        */

        /**
        * Current Prime Time multiplier (eg. if primetime_percentage is 100%, primetime_multiplier will be 2.0)
        * @member {Number} primetime_multiplier
        */

        /**
        * A token that confirms the user has accepted current Prime Time charges (Deprecated)
        * @member {String} primetime_confirmation_token
        */

        /**
        * A token that confirms the user has accepted current Prime Time and/or fixed price charges
        * @member {String} cost_token
        */

        /**
        * Validity of the token in seconds
        * @member {String} token_duration
        */

    }]);

    return RideRequestError;
}();

exports.default = RideRequestError;
