// node-lyft/lib/model/RideTypeEnumWithOther.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require("../ApiClient");

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* Enum class RideTypeEnumWithOther.
* @enum {}
* @readonly
*/
var RideTypeEnumWithOther = function () {
    function RideTypeEnumWithOther() {
        _classCallCheck(this, RideTypeEnumWithOther);

        this.lyft = "lyft";
        this.lyft_line = "lyft_line";
        this.lyft_plus = "lyft_plus";
        this.lyft_premier = "lyft_premier";
        this.lyft_lux = "lyft_lux";
        this.lyft_luxsuv = "lyft_luxsuv";
        this.other = "other";
    }

    /**
     * value: "lyft"
     * @const
     */


    /**
     * value: "lyft_line"
     * @const
     */


    /**
     * value: "lyft_plus"
     * @const
     */


    /**
     * value: "lyft_premier"
     * @const
     */


    /**
     * value: "lyft_lux"
     * @const
     */


    /**
     * value: "lyft_luxsuv"
     * @const
     */


    /**
     * value: "other"
     * @const
     */


    _createClass(RideTypeEnumWithOther, null, [{
        key: "constructFromObject",


        /**
        * Returns a <code>RideTypeEnumWithOther</code> enum value from a Javascript object name.
        * @param {Object} data The plain JavaScript object containing the name of the enum value.
        * @return {module:model/RideTypeEnumWithOther} The enum <code>RideTypeEnumWithOther</code> value.
        */
        value: function constructFromObject(object) {
            return object;
        }
    }]);

    return RideTypeEnumWithOther;
}();

exports.default = RideTypeEnumWithOther;
