// node-lyft/lib/model/RideReceipt.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _Charge = require('./Charge');

var _Charge2 = _interopRequireDefault(_Charge);

var _LineItem = require('./LineItem');

var _LineItem2 = _interopRequireDefault(_LineItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The RideReceipt model module.
* @module model/RideReceipt
* @version 1.0.0
*/
var RideReceipt = function () {
    /**
    * Constructs a new <code>RideReceipt</code>.
    * Receipt information of a processed ride.
    * @alias module:model/RideReceipt
    * @class
    */

    function RideReceipt() {
        _classCallCheck(this, RideReceipt);

        this.ride_id = undefined;
        this.price = undefined;
        this.line_items = undefined;
        this.charges = undefined;
        this.requested_at = undefined;
        this.ride_profile = undefined;
    }

    /**
    * Constructs a <code>RideReceipt</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/RideReceipt} obj Optional instance to populate.
    * @return {module:model/RideReceipt} The populated <code>RideReceipt</code> instance.
    */


    _createClass(RideReceipt, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new RideReceipt();

                if (data.hasOwnProperty('ride_id')) {
                    obj['ride_id'] = _ApiClient2.default.convertToType(data['ride_id'], 'String');
                }
                if (data.hasOwnProperty('price')) {
                    obj['price'] = _ApiClient2.default.convertToType(data['price'], Object);
                }
                if (data.hasOwnProperty('line_items')) {
                    obj['line_items'] = _ApiClient2.default.convertToType(data['line_items'], [_LineItem2.default]);
                }
                if (data.hasOwnProperty('charges')) {
                    obj['charges'] = _ApiClient2.default.convertToType(data['charges'], [_Charge2.default]);
                }
                if (data.hasOwnProperty('requested_at')) {
                    obj['requested_at'] = _ApiClient2.default.convertToType(data['requested_at'], 'Date');
                }
                if (data.hasOwnProperty('ride_profile')) {
                    obj['ride_profile'] = _ApiClient2.default.convertToType(data['ride_profile'], Object);
                }
            }
            return obj;
        }

        /**
        * The unique ID of this ride
        * @member {String} ride_id
        */

        /**
        * The total price for the current ride
        * @member {Object} price
        */

        /**
        * The break down of line items
        * @member {Array.<module:model/LineItem>} line_items
        */

        /**
        * The break down of charge method
        * @member {Array.<module:model/Charge>} charges
        */

        /**
        * The ride requested timestamp in date and time
        * @member {Date} requested_at
        */

        /**
        * Indicates whether the ride was requested from the business profile or personal profile of the user. 
        * @member {Object} ride_profile
        */

    }]);

    return RideReceipt;
}();

exports.default = RideReceipt;
