// node-lyft/lib/model/RideDetail.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _DriverDetail = require('./DriverDetail');

var _DriverDetail2 = _interopRequireDefault(_DriverDetail);

var _LineItem = require('./LineItem');

var _LineItem2 = _interopRequireDefault(_LineItem);

var _PassengerDetail = require('./PassengerDetail');

var _PassengerDetail2 = _interopRequireDefault(_PassengerDetail);

var _RideStatusEnum = require('./RideStatusEnum');

var _RideStatusEnum2 = _interopRequireDefault(_RideStatusEnum);

var _RideTypeEnumWithOther = require('./RideTypeEnumWithOther');

var _RideTypeEnumWithOther2 = _interopRequireDefault(_RideTypeEnumWithOther);

var _VehicleDetail = require('./VehicleDetail');

var _VehicleDetail2 = _interopRequireDefault(_VehicleDetail);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The RideDetail model module.
* @module model/RideDetail
* @version 1.0.0
*/
var RideDetail = function () {
    /**
    * Constructs a new <code>RideDetail</code>.
    * Detail information about a ride
    * @alias module:model/RideDetail
    * @class
    */

    function RideDetail() {
        _classCallCheck(this, RideDetail);

        this.ride_id = undefined;
        this.status = undefined;
        this.ride_type = undefined;
        this.passenger = undefined;
        this.driver = undefined;
        this.vehicle = undefined;
        this.origin = undefined;
        this.destination = undefined;
        this.pickup = undefined;
        this.dropoff = undefined;
        this.location = undefined;
        this.primetime_percentage = undefined;
        this.distance_miles = undefined;
        this.duration_seconds = undefined;
        this.price = undefined;
        this.line_items = undefined;
        this.can_cancel = undefined;
        this.canceled_by = undefined;
        this.cancellation_price = undefined;
        this.rating = undefined;
        this.feedback = undefined;
        this.pricing_details_url = undefined;
        this.route_url = undefined;
        this.requested_at = undefined;
        this.generated_at = undefined;
        this.ride_profile = undefined;
        this.beacon_color = undefined;
    }

    /**
    * Constructs a <code>RideDetail</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/RideDetail} obj Optional instance to populate.
    * @return {module:model/RideDetail} The populated <code>RideDetail</code> instance.
    */


    _createClass(RideDetail, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new RideDetail();

                if (data.hasOwnProperty('ride_id')) {
                    obj['ride_id'] = _ApiClient2.default.convertToType(data['ride_id'], 'String');
                }
                if (data.hasOwnProperty('status')) {
                    obj['status'] = _RideStatusEnum2.default.constructFromObject(data['status']);
                }
                if (data.hasOwnProperty('ride_type')) {
                    obj['ride_type'] = _RideTypeEnumWithOther2.default.constructFromObject(data['ride_type']);
                }
                if (data.hasOwnProperty('passenger')) {
                    obj['passenger'] = _PassengerDetail2.default.constructFromObject(data['passenger']);
                }
                if (data.hasOwnProperty('driver')) {
                    obj['driver'] = _DriverDetail2.default.constructFromObject(data['driver']);
                }
                if (data.hasOwnProperty('vehicle')) {
                    obj['vehicle'] = _VehicleDetail2.default.constructFromObject(data['vehicle']);
                }
                if (data.hasOwnProperty('origin')) {
                    obj['origin'] = _ApiClient2.default.convertToType(data['origin'], Object);
                }
                if (data.hasOwnProperty('destination')) {
                    obj['destination'] = _ApiClient2.default.convertToType(data['destination'], Object);
                }
                if (data.hasOwnProperty('pickup')) {
                    obj['pickup'] = _ApiClient2.default.convertToType(data['pickup'], Object);
                }
                if (data.hasOwnProperty('dropoff')) {
                    obj['dropoff'] = _ApiClient2.default.convertToType(data['dropoff'], Object);
                }
                if (data.hasOwnProperty('location')) {
                    obj['location'] = _ApiClient2.default.convertToType(data['location'], Object);
                }
                if (data.hasOwnProperty('primetime_percentage')) {
                    obj['primetime_percentage'] = _ApiClient2.default.convertToType(data['primetime_percentage'], 'String');
                }
                if (data.hasOwnProperty('distance_miles')) {
                    obj['distance_miles'] = _ApiClient2.default.convertToType(data['distance_miles'], 'Number');
                }
                if (data.hasOwnProperty('duration_seconds')) {
                    obj['duration_seconds'] = _ApiClient2.default.convertToType(data['duration_seconds'], 'Number');
                }
                if (data.hasOwnProperty('price')) {
                    obj['price'] = _ApiClient2.default.convertToType(data['price'], Object);
                }
                if (data.hasOwnProperty('line_items')) {
                    obj['line_items'] = _ApiClient2.default.convertToType(data['line_items'], [_LineItem2.default]);
                }
                if (data.hasOwnProperty('can_cancel')) {
                    obj['can_cancel'] = _ApiClient2.default.convertToType(data['can_cancel'], ['String']);
                }
                if (data.hasOwnProperty('canceled_by')) {
                    obj['canceled_by'] = _ApiClient2.default.convertToType(data['canceled_by'], 'String');
                }
                if (data.hasOwnProperty('cancellation_price')) {
                    obj['cancellation_price'] = _ApiClient2.default.convertToType(data['cancellation_price'], Object);
                }
                if (data.hasOwnProperty('rating')) {
                    obj['rating'] = _ApiClient2.default.convertToType(data['rating'], 'Number');
                }
                if (data.hasOwnProperty('feedback')) {
                    obj['feedback'] = _ApiClient2.default.convertToType(data['feedback'], 'String');
                }
                if (data.hasOwnProperty('pricing_details_url')) {
                    obj['pricing_details_url'] = _ApiClient2.default.convertToType(data['pricing_details_url'], 'String');
                }
                if (data.hasOwnProperty('route_url')) {
                    obj['route_url'] = _ApiClient2.default.convertToType(data['route_url'], 'String');
                }
                if (data.hasOwnProperty('requested_at')) {
                    obj['requested_at'] = _ApiClient2.default.convertToType(data['requested_at'], 'Date');
                }
                if (data.hasOwnProperty('generated_at')) {
                    obj['generated_at'] = _ApiClient2.default.convertToType(data['generated_at'], 'Date');
                }
                if (data.hasOwnProperty('ride_profile')) {
                    obj['ride_profile'] = _ApiClient2.default.convertToType(data['ride_profile'], Object);
                }
                if (data.hasOwnProperty('beacon_color')) {
                    obj['beacon_color'] = _ApiClient2.default.convertToType(data['beacon_color'], 'String');
                }
            }
            return obj;
        }

        /**
        * The unique ID of this ride
        * @member {String} ride_id
        */

        /**
        * @member {module:model/RideStatusEnum} status
        */

        /**
        * @member {module:model/RideTypeEnumWithOther} ride_type
        */

        /**
        * @member {module:model/PassengerDetail} passenger
        */

        /**
        * @member {module:model/DriverDetail} driver
        */

        /**
        * @member {module:model/VehicleDetail} vehicle
        */

        /**
        * The *requested* location for passenger pickup
        * @member {Object} origin
        */

        /**
        * The *requested* location for passenger drop off
        * @member {Object} destination
        */

        /**
        * The *actual* location of passenger pickup
        * @member {Object} pickup
        */

        /**
        * The *actual* location of passenger drop off
        * @member {Object} dropoff
        */

        /**
        * The *current* location info of the ride
        * @member {Object} location
        */

        /**
        * The Prime Time percentage applied to the base price
        * @member {String} primetime_percentage
        */

        /**
        * The distance, in miles, that this ride traveled. This field is only present after drop-off
        * @member {Number} distance_miles
        */

        /**
        * Duration of the ride in seconds from pickup to drop-off. This field is only present after drop-off.
        * @member {Number} duration_seconds
        */

        /**
        * The total price for the current ride
        * @member {Object} price
        */

        /**
        * The break down of cost
        * @member {Array.<module:model/LineItem>} line_items
        */

        /**
        * @member {Array.<module:model/RideDetail.CanCancelEnum>} can_cancel
        */

        /**
        * The role of user who canceled the ride (if applicable)
        * @member {String} canceled_by
        */

        /**
        * The cost of cancellation if there would be a penalty
        * @member {Object} cancellation_price
        */

        /**
        * The rating the user left for this ride, from 1 to 5
        * @member {Number} rating
        */

        /**
        * The written feedback the user left for this ride
        * @member {String} feedback
        */

        /**
        * The web view showing the pricing structure for the geographic area where the ride was taken 
        * @member {String} pricing_details_url
        */

        /**
        * The web view showing the passenger, driver, and route for this ride. This field will only be present for rides created through this API, or that have been shared through the \"Share my Route\" feature 
        * @member {String} route_url
        */

        /**
        * The ride requested timestamp in date and time
        * @member {Date} requested_at
        */

        /**
        * The request timestamp in date and time
        * @member {Date} generated_at
        */

        /**
        * Indicates whether the ride was requested from the business profile or personal profile of the user. 
        * @member {Object} ride_profile
        */

        /**
        * Hex color code of the driver AMP device.
        * @member {String} beacon_color
        */


        /**
        * Allowed values for the <code>canCancel</code> property.
        * @enum {String}
        * @readonly
        */

    }]);

    return RideDetail;
}();

RideDetail.CanCancelEnum = {

    /**
     * value: "driver"
     * @const
     */
    "driver": "driver",

    /**
     * value: "passenger"
     * @const
     */
    "passenger": "passenger",

    /**
     * value: "dispatcher"
     * @const
     */
    "dispatcher": "dispatcher"
};
exports.default = RideDetail;
