// node-lyft/lib/model/RideRequest.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _PassengerDetail = require('./PassengerDetail');

var _PassengerDetail2 = _interopRequireDefault(_PassengerDetail);

var _RideStatusEnum = require('./RideStatusEnum');

var _RideStatusEnum2 = _interopRequireDefault(_RideStatusEnum);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The RideRequest model module.
* @module model/RideRequest
* @version 1.0.0
*/
var RideRequest = function () {
    /**
    * Constructs a new <code>RideRequest</code>.
    * Minimal set of ride details
    * @alias module:model/RideRequest
    * @class
    */

    function RideRequest() {
        _classCallCheck(this, RideRequest);

        this.ride_id = undefined;
        this.status = undefined;
        this.origin = undefined;
        this.destination = undefined;
        this.passenger = undefined;
    }

    /**
    * Constructs a <code>RideRequest</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/RideRequest} obj Optional instance to populate.
    * @return {module:model/RideRequest} The populated <code>RideRequest</code> instance.
    */


    _createClass(RideRequest, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new RideRequest();

                if (data.hasOwnProperty('ride_id')) {
                    obj['ride_id'] = _ApiClient2.default.convertToType(data['ride_id'], 'String');
                }
                if (data.hasOwnProperty('status')) {
                    obj['status'] = _RideStatusEnum2.default.constructFromObject(data['status']);
                }
                if (data.hasOwnProperty('origin')) {
                    obj['origin'] = _ApiClient2.default.convertToType(data['origin'], Object);
                }
                if (data.hasOwnProperty('destination')) {
                    obj['destination'] = _ApiClient2.default.convertToType(data['destination'], Object);
                }
                if (data.hasOwnProperty('passenger')) {
                    obj['passenger'] = _PassengerDetail2.default.constructFromObject(data['passenger']);
                }
            }
            return obj;
        }

        /**
        * The ID of the requested ride
        * @member {String} ride_id
        */

        /**
        * @member {module:model/RideStatusEnum} status
        */

        /**
        * The *requested* location for passenger pickup
        * @member {Object} origin
        */

        /**
        * The *requested* location for passenger drop off
        * @member {Object} destination
        */

        /**
        * @member {module:model/PassengerDetail} passenger
        */

    }]);

    return RideRequest;
}();

exports.default = RideRequest;
