// node-lyft/lib/model/RideType.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _PricingDetails = require('./PricingDetails');

var _PricingDetails2 = _interopRequireDefault(_PricingDetails);

var _RideTypeEnum = require('./RideTypeEnum');

var _RideTypeEnum2 = _interopRequireDefault(_RideTypeEnum);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The RideType model module.
* @module model/RideType
* @version 1.0.0
*/
var RideType = function () {
    /**
    * Constructs a new <code>RideType</code>.
    * @alias module:model/RideType
    * @class
    */

    function RideType() {
        _classCallCheck(this, RideType);

        this.ride_type = undefined;
        this.display_name = undefined;
        this.seats = undefined;
        this.image_url = undefined;
        this.pricing_details = undefined;
        this.scheduled_pricing_details = undefined;
    }

    /**
    * Constructs a <code>RideType</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/RideType} obj Optional instance to populate.
    * @return {module:model/RideType} The populated <code>RideType</code> instance.
    */


    _createClass(RideType, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new RideType();

                if (data.hasOwnProperty('ride_type')) {
                    obj['ride_type'] = _RideTypeEnum2.default.constructFromObject(data['ride_type']);
                }
                if (data.hasOwnProperty('display_name')) {
                    obj['display_name'] = _ApiClient2.default.convertToType(data['display_name'], 'String');
                }
                if (data.hasOwnProperty('seats')) {
                    obj['seats'] = _ApiClient2.default.convertToType(data['seats'], 'Number');
                }
                if (data.hasOwnProperty('image_url')) {
                    obj['image_url'] = _ApiClient2.default.convertToType(data['image_url'], 'String');
                }
                if (data.hasOwnProperty('pricing_details')) {
                    obj['pricing_details'] = _PricingDetails2.default.constructFromObject(data['pricing_details']);
                }
                if (data.hasOwnProperty('scheduled_pricing_details')) {
                    obj['scheduled_pricing_details'] = _PricingDetails2.default.constructFromObject(data['scheduled_pricing_details']);
                }
            }
            return obj;
        }

        /**
        * @member {module:model/RideTypeEnum} ride_type
        */

        /**
        * A human readable description of the ride type
        * @member {String} display_name
        */

        /**
        * The maximum number of seats available for rides requested with this ride type
        * @member {Number} seats
        */

        /**
        * The URL of an image representing this ride type
        * @member {String} image_url
        */

        /**
        * @member {module:model/PricingDetails} pricing_details
        */

        /**
        * @member {module:model/PricingDetails} scheduled_pricing_details
        */

    }]);

    return RideType;
}();

exports.default = RideType;
