// node-lyft/lib/model/SandboxDriverAvailability.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The SandboxDriverAvailability model module.
* @module model/SandboxDriverAvailability
* @version 1.0.0
*/
var SandboxDriverAvailability = function () {
    /**
    * Constructs a new <code>SandboxDriverAvailability</code>.
    * @alias module:model/SandboxDriverAvailability
    * @class
    * @param lat {Number} The latitude component of a location
    * @param lng {Number} The longitude component of a location
    * @param driverAvailability {Boolean} The availability of driver in a region
    */

    function SandboxDriverAvailability(lat, lng, driverAvailability) {
        _classCallCheck(this, SandboxDriverAvailability);

        this.lat = undefined;
        this.lng = undefined;
        this.driver_availability = undefined;


        this['lat'] = lat;this['lng'] = lng;this['driver_availability'] = driverAvailability;
    }

    /**
    * Constructs a <code>SandboxDriverAvailability</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/SandboxDriverAvailability} obj Optional instance to populate.
    * @return {module:model/SandboxDriverAvailability} The populated <code>SandboxDriverAvailability</code> instance.
    */


    _createClass(SandboxDriverAvailability, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new SandboxDriverAvailability();

                if (data.hasOwnProperty('lat')) {
                    obj['lat'] = _ApiClient2.default.convertToType(data['lat'], 'Number');
                }
                if (data.hasOwnProperty('lng')) {
                    obj['lng'] = _ApiClient2.default.convertToType(data['lng'], 'Number');
                }
                if (data.hasOwnProperty('driver_availability')) {
                    obj['driver_availability'] = _ApiClient2.default.convertToType(data['driver_availability'], 'Boolean');
                }
            }
            return obj;
        }

        /**
        * The latitude component of a location
        * @member {Number} lat
        */

        /**
        * The longitude component of a location
        * @member {Number} lng
        */

        /**
        * The availability of driver in a region
        * @member {Boolean} driver_availability
        */

    }]);

    return SandboxDriverAvailability;
}();

exports.default = SandboxDriverAvailability;
