// node-lyft/lib/model/SandboxRideType.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _RideTypeEnum = require('./RideTypeEnum');

var _RideTypeEnum2 = _interopRequireDefault(_RideTypeEnum);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The SandboxRideType model module.
* @module model/SandboxRideType
* @version 1.0.0
*/
var SandboxRideType = function () {
    /**
    * Constructs a new <code>SandboxRideType</code>.
    * @alias module:model/SandboxRideType
    * @class
    * @param lat {Number} The latitude component of a location
    * @param lng {Number} The longitude component of a location
    * @param rideTypes {Array.<module:model/RideTypeEnum>} 
    */

    function SandboxRideType(lat, lng, rideTypes) {
        _classCallCheck(this, SandboxRideType);

        this.lat = undefined;
        this.lng = undefined;
        this.ride_types = undefined;


        this['lat'] = lat;this['lng'] = lng;this['ride_types'] = rideTypes;
    }

    /**
    * Constructs a <code>SandboxRideType</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/SandboxRideType} obj Optional instance to populate.
    * @return {module:model/SandboxRideType} The populated <code>SandboxRideType</code> instance.
    */


    _createClass(SandboxRideType, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new SandboxRideType();

                if (data.hasOwnProperty('lat')) {
                    obj['lat'] = _ApiClient2.default.convertToType(data['lat'], 'Number');
                }
                if (data.hasOwnProperty('lng')) {
                    obj['lng'] = _ApiClient2.default.convertToType(data['lng'], 'Number');
                }
                if (data.hasOwnProperty('ride_types')) {
                    obj['ride_types'] = _ApiClient2.default.convertToType(data['ride_types'], [_RideTypeEnum2.default]);
                }
            }
            return obj;
        }

        /**
        * The latitude component of a location
        * @member {Number} lat
        */

        /**
        * The longitude component of a location
        * @member {Number} lng
        */

        /**
        * @member {Array.<module:model/RideTypeEnum>} ride_types
        */

    }]);

    return SandboxRideType;
}();

exports.default = SandboxRideType;
