// node-lyft/lib/model/UserDetail.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The UserDetail model module.
* @module model/UserDetail
* @version 1.0.0
*/
var UserDetail = function () {
    /**
    * Constructs a new <code>UserDetail</code>.
    * @alias module:model/UserDetail
    * @class
    * @param firstName {String} The passenger's first name
    * @param imageUrl {String} The passenger's profile image
    * @param rating {String} The passenger's rating
    */

    function UserDetail(firstName, imageUrl, rating) {
        _classCallCheck(this, UserDetail);

        this.first_name = undefined;
        this.image_url = undefined;
        this.rating = undefined;


        this['first_name'] = firstName;this['image_url'] = imageUrl;this['rating'] = rating;
    }

    /**
    * Constructs a <code>UserDetail</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/UserDetail} obj Optional instance to populate.
    * @return {module:model/UserDetail} The populated <code>UserDetail</code> instance.
    */


    _createClass(UserDetail, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new UserDetail();

                if (data.hasOwnProperty('first_name')) {
                    obj['first_name'] = _ApiClient2.default.convertToType(data['first_name'], 'String');
                }
                if (data.hasOwnProperty('image_url')) {
                    obj['image_url'] = _ApiClient2.default.convertToType(data['image_url'], 'String');
                }
                if (data.hasOwnProperty('rating')) {
                    obj['rating'] = _ApiClient2.default.convertToType(data['rating'], 'String');
                }
            }
            return obj;
        }

        /**
        * The passenger's first name
        * @member {String} first_name
        */

        /**
        * The passenger's profile image
        * @member {String} image_url
        */

        /**
        * The passenger's rating
        * @member {String} rating
        */

    }]);

    return UserDetail;
}();

exports.default = UserDetail;
