// node-lyft/lib/model/VehicleDetail.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Lyft API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Drive your app to success with Lyft's API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * OpenAPI spec version: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: api-support@lyft.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by the swagger code generator program.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://github.com/swagger-api/swagger-codegen.git
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
* The VehicleDetail model module.
* @module model/VehicleDetail
* @version 1.0.0
*/
var VehicleDetail = function () {
    /**
    * Constructs a new <code>VehicleDetail</code>.
    * @alias module:model/VehicleDetail
    * @class
    * @param make {String} The vehicle's maker
    * @param model {String} The vehicle's model
    * @param year {Number} The vehicle's model year
    * @param licensePlate {String} The vehicle's license plate
    * @param licensePlateState {String} The vehicle's license plate state
    * @param color {String} The vehicle's color
    * @param imageUrl {String} The vehicle's image url
    */

    function VehicleDetail(make, model, year, licensePlate, licensePlateState, color, imageUrl) {
        _classCallCheck(this, VehicleDetail);

        this.make = undefined;
        this.model = undefined;
        this.year = undefined;
        this.license_plate = undefined;
        this.license_plate_state = undefined;
        this.color = undefined;
        this.image_url = undefined;


        this['make'] = make;this['model'] = model;this['year'] = year;this['license_plate'] = licensePlate;this['license_plate_state'] = licensePlateState;this['color'] = color;this['image_url'] = imageUrl;
    }

    /**
    * Constructs a <code>VehicleDetail</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/VehicleDetail} obj Optional instance to populate.
    * @return {module:model/VehicleDetail} The populated <code>VehicleDetail</code> instance.
    */


    _createClass(VehicleDetail, null, [{
        key: 'constructFromObject',
        value: function constructFromObject(data, obj) {
            if (data) {
                obj = obj || new VehicleDetail();

                if (data.hasOwnProperty('make')) {
                    obj['make'] = _ApiClient2.default.convertToType(data['make'], 'String');
                }
                if (data.hasOwnProperty('model')) {
                    obj['model'] = _ApiClient2.default.convertToType(data['model'], 'String');
                }
                if (data.hasOwnProperty('year')) {
                    obj['year'] = _ApiClient2.default.convertToType(data['year'], 'Number');
                }
                if (data.hasOwnProperty('license_plate')) {
                    obj['license_plate'] = _ApiClient2.default.convertToType(data['license_plate'], 'String');
                }
                if (data.hasOwnProperty('license_plate_state')) {
                    obj['license_plate_state'] = _ApiClient2.default.convertToType(data['license_plate_state'], 'String');
                }
                if (data.hasOwnProperty('color')) {
                    obj['color'] = _ApiClient2.default.convertToType(data['color'], 'String');
                }
                if (data.hasOwnProperty('image_url')) {
                    obj['image_url'] = _ApiClient2.default.convertToType(data['image_url'], 'String');
                }
            }
            return obj;
        }

        /**
        * The vehicle's maker
        * @member {String} make
        */

        /**
        * The vehicle's model
        * @member {String} model
        */

        /**
        * The vehicle's model year
        * @member {Number} year
        */

        /**
        * The vehicle's license plate
        * @member {String} license_plate
        */

        /**
        * The vehicle's license plate state
        * @member {String} license_plate_state
        */

        /**
        * The vehicle's color
        * @member {String} color
        */

        /**
        * The vehicle's image url
        * @member {String} image_url
        */

    }]);

    return VehicleDetail;
}();

exports.default = VehicleDetail;
